### PR TITLE
Configuration drawers refactor

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

Code refactor to better share code across create and reconfigure

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/v2/shared/FundingCycleConfigurationDrawers/__tests__/FundingCycleDrawer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  describe('useFundingCycleDrawer', () => {
+    const instanceUnderTest = (onClose: VoidFunction = () => {}) => {
+      const { result } = renderHook(() => useFundingCycleDrawer(onClose))
+      return result
+    }
+    test('calls onClose when emitDrawerClose is called', () => {
+      let called = false
+      const result = instanceUnderTest(() => (called = true))
+      result.current.emitDrawerClose()
+      expect(called).toEqual(true)
```
Comment (aeolianeth) [inline src/components/v2/shared/FundingCycleConfigurationDrawers/__tests__/FundingCycleDrawer.test.tsx:15]: an alternative might be (not sure if it works lol):

```suggestion
      const callback = jest.fn()
      const result = instanceUnderTest(callback)
      result.current.emitDrawerClose()
      expect(callback).toHaveBeenCalled()
```

File: src/components/v2/shared/FundingCycleConfigurationDrawers/TokenDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  onClose: VoidFunction
+  isCreate?: boolean
+}) {
+  const {
+    handleDrawerCloseClick,
+    emitDrawerClose,
+    setFormUpdated,
+    unsavedChangesModalVisible,
+    closeModal,
+  } = useFundingCycleDrawer(onClose)
```
Comment (aeolianeth) [inline src/components/v2/shared/FundingCycleConfigurationDrawers/TokenDrawer.tsx:23]: fucking fresh ser. Such a good approach. We need to do more of this probably.

cc @johnnyd-eth @peripheralist

File: src/components/v2/shared/FundingCycleConfigurationDrawers/FundingCycleDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Provides the blueprint for Funding Cycle Drawers.
+ */
+export default function FundingCycleDrawer({
+  title,
+  visible,
+  onClose,
+  children,
+}: PropsWithChildren<{
```
Comment (aeolianeth) [inline src/components/v2/shared/FundingCycleConfigurationDrawers/FundingCycleDrawer.tsx:14]: **praise** ty for using `PropsWithChildren`

**Review by aeolianeth — COMMENTED** [review]
@wraeth-eth real real nice. Love to see some refactoring happening, and a test to boot!

Just gonna give this a smoke test, if all good, will merge.

**Comment by aeolianeth** [comment]
@wraeth-eth heads up - rebasing

File: src/components/v2/shared/FundingCycleConfigurationDrawers/TokenDrawer.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/v2/shared/FundingCycleConfigurationDrawers/TokenDrawer.tsx:23]: Yeah nice love it

File: src/components/v2/shared/FundingCycleConfigurationDrawers/__tests__/FundingCycleDrawer.test.tsx (hunk 1 above)
Comment (wraeth-eth) [reply src/components/v2/shared/FundingCycleConfigurationDrawers/__tests__/FundingCycleDrawer.test.tsx:15]: TIFL 🙌

File: src/components/v2/shared/FundingCycleConfigurationDrawers/__tests__/FundingCycleDrawer.test.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import { act, renderHook } from '@testing-library/react-hooks'
+
+import { useFundingCycleDrawer } from '../useFundingCycleDrawer'
+
+describe('FundingCycleDrawer', () => {
+  describe('useFundingCycleDrawer', () => {
+    const instanceUnderTest = (onClose: VoidFunction = () => {}) => {
+      const { result } = renderHook(() => useFundingCycleDrawer(onClose))
```
[the remark this answers is not in the document]
Comment (wraeth-eth) [reply src/components/v2/shared/FundingCycleConfigurationDrawers/__tests__/FundingCycleDrawer.test.tsx:8]: I need to get up to speed on testing ui in general - so used to abstracting service code to run tests with mocks lol

**Pull request merged**